core/templates/dev/head/domain/editor/UndoRedoService.js

```diff
@@ -114,11 +125,25 @@ var UndoRedoService = function() {
 
   /**
    * Clears the change history. This does not reverse any of the changes applied
-   * from applyChange() or redoChange().
+   * from applyChange() or redoChange(). This will invoke the callback provided
+   * to setOnChangedCallback(), if one is available.
    */
   UndoRedoService.clearChanges = function() {
     _appliedChanges = [];
     _undoneChanges = [];
+    _dispatchMutation(null);
+  };
+
+  /**
+   * Sets a callback which will be called whenever the UndoRedoService changes.
+   * Please note that the callback is only called after a change is applied. The
+   * function may take two arguments, where the first one is a change being
+   * applied and the second is a boolean on whether that change is being applied
+   * or reverted. The change object passed to the callback will be null if the
+   * UndoRedoService is being cleared of changes.
+   */
+  UndoRedoService.setOnChangedCallback = function(callback) {
+    _mutationCallback = callback;
```

Comment:
Should this be immutable once set? If not, then the docstrings above should probably refer to "the latest callback provided to setOnChangeCallback()".